### PR TITLE
CBBD-89 developed code for Carrier RIF parsing

### PR DESCRIPTION
**Pull request by dho937**
@karlmdavis @sbrockhoff   Please review when you get a chance.  I removed all of the .ordinal() coding from the other resources.  Thanks.

### CODE REVIEW

**Comment by karlmdavis** [comment]
@dho937: Some initial feedback:

1. Looks like the build is failing with a compile error on `DataTransformerTest`.
1. In the future, please try to keep large refactorings in separate commits from enhancements or fixes. This one is going to be a bit painful to review, with them being mixed together like that.

**Comment by dho937** [comment]
@sbrockhoff @karlmdavis   Fixed the DataTransformerTest class.

**Comment by sbrockhoff** [comment]
:+1:

File: bluebutton-data-pipeline-rif/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/model/CarrierClaimGroup.java [hunk 1]
```diff
@@ -31,6 +31,10 @@
  */
 public final class CarrierClaimGroup {
 	/**
+	 * @see java.lang.Object#toString()
+	 */
+
+	/**
```
Comment (karlmdavis) [inline bluebutton-data-pipeline-rif/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/model/CarrierClaimGroup.java]: This looks wrong: what is that JavaDoc attached to?

File: bluebutton-data-pipeline-rif/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/model/CarrierClaimGroup.java [hunk 2]
```diff
@@ -836,21 +1242,24 @@ public String toString() {
...
+		 * <a href=
 		 * "https://www.ccwdata.org/cs/groups/public/documents/datadictionary/lnndccd.txt">
 		 * CCW Data Dictionary: LNNDCCD</a>.
 		 */
 		LINE_NDC_CD,
 
 		/**
-		 * Type: <code>CHAR</code>, max chars: 10. See <a href=
+		 * Type: <code>CHAR</code>, max chars: 10 <code>Optional</code>. See
+		 * <a href=
```
Comment (karlmdavis) [inline bluebutton-data-pipeline-rif/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/model/CarrierClaimGroup.java]: Fantastic: thanks for including the "Optional" indications!

**Comment by karlmdavis** [comment]
:+1: 

Though I'll suggest that, going forwards, it'd be better for us to tackle fewer fields in one go, to allow us to instead parse **and** transform them. For instance, and I haven't even looked at the docs for it, but I suspect that the `hpsaScarcityCode` doesn't map to anything useful to benes, and doesn't fit in FHIR. No point in parsing fields that we won't transform.

**Pull request merged**